src/gui/modals/ExportProjectDialog.cpp

```diff
@@ -247,10 +270,31 @@ void ExportProjectDialog::onFileFormatChanged(int index)
 	checkBoxVariableBitRate->setVisible(variableBitrateVisible);
 
 	depthWidget->setVisible(bitDepthControlEnabled);
+		
+	// Updating target file extention and checking status
+
+	m_targetFileInfo.setFile(removeKnownExtension(m_targetFileInfo.absoluteFilePath()) 
+							+ ProjectRenderer::fileEncodeDevices[index].m_extension);
+	
+	m_needCheckIfTargetFileExists = m_originalTargetFileExt.compare(ProjectRenderer::fileEncodeDevices[index].m_extension, 
+																	Qt::CaseInsensitive) != 0;
```

Comment:
Is a case-insensitive comparison correct here? On Windows it's probably the right choice, but on platforms with case-sensitive file systems, you could end up overwriting a file unintentionally if your original choice of file name differed from an existing file only in the case of the extension. It seems like an unlikely scenario, but it may be better to play it safe.

Reply:
But here if it to check if the suffix is same, even with different case. And then checking would be done. So seems to be protected.

Reply:
If the extension is the same, ignoring case, then no checking is done. The checking is done when the extension is different, and the concept of "different" ought to match that of the underlying file system. If not, the safe option is to be case sensitive. Extra checking can't hurt (besides being a minor annoyance to the user), whereas missing a check could cause data loss.

Reply:
So maybe we should check always in the second window (the real one doing the export) and not from file selector.

Reply:
That sounds like a good idea. Perhaps you could even show the file dialog after the start button is pressed, rather than before the export dialog. That way you can have the dialog only show appropriate file extensions, and don't need to worry about changing it automatically.